### ISSUE TITLE
Errors for small LZ4 compressed files

### ISSUE THREAD

**Issue by bcyran** [Genivia/ugrep#214]
`ugrep` fails when searching through small LZ4 compressed files. Reproduction:
```shell
/tmp/lz4 ❯ echo "this file is small" > small.txt
/tmp/lz4 ❯ echo "this file is bigger\nthis file is bigger" > bigger.txt
/tmp/lz4 ❯ lz4 -m small.txt bigger.txt                                                                                                                                                                                
/tmp/lz4 ❯ ugrep -z file -- *.lz4
ugrep: warning: cannot decompress small.txt.lz4: an error was detected in the lz4 compressed data
bigger.txt.lz4:this file is bigger
bigger.txt.lz4:this file is bigger
```

The compressed files are not broken in any way which is confirmed by `lz4 --test`, and by the fact that they can be properly decompressed.

I'm not able to specify what "small" means. I was not able to find threshold value of  file size before compression or after compression. However it doesn't seem to occur for files which are bigger than a few dozen bytes.

I tried to play a little with `lz4` options to check if it affects the issue, but wasn't able to find anything.

**Comment by genivia-inc** [comment]
Got it. This was trivial to fix. Will update ugrep soon. The _uncompressed data bit_ was accidentally ignored in the [lz4 frame](https://fuchsia.googlesource.com/third_party/lz4/+/HEAD/doc/lz4_Frame_format.md). Very small files are stored uncompressed. This was apparently missed on our end when testing ugrep with lz4, despite our extensive testing and reviewing. I wished their repo included test files, but it only has scripts to test lz4 itself.

**Comment by genivia-inc** [comment]
Fixed in v3.8.3.
